documentation/modules/non-admin-permissions-for-ui.adoc

```diff
@@ -0,0 +1,56 @@
+:_content-type: PROCEDURE
+
+[id="non-admin-permissions_{context}"]
+= Granting permissions to non-administrators to work with migration plan components
+
+As an administrator, you can work with all components of migration plans (for example, providers, network mappings, and migration plans).
+
+By default, non-administrators have limited ability to work with migration plans and their components. You can modify their roles to allow them full access to all components or you can give them limited permissions.
+
+For example, you can grant non-administrators the following set of permissions:
+
+* Create and modify storage maps, network maps, and migration plans for the namespaces they have access to.
+* Attach providers created by adminstrators to storage maps, network maps, and migration plans.
+
+[cols="3", options="header"]
+.Permissions required for non-adminstrators to work with migration plan components but not create providers
+|===
+|Actions |API groups |Resources
+
+|All
+|`forklift.konveyer.io`
+|`plans`
+
+|All
+|`forklift.konveyer.io`
+|`migrations`
+
+|All
+|`forklift.konveyer.io`
+|`hooks`
+
+|`get`, `list`, `watch`
+|`forklift.konveyer.io`
+|`providers`
+
+|All
+|`forklift.konveyer.io`
+|`networkmaps`
+
+|All
+|`forklift.konveyer.io`
+|`storagemaps`
+
+|`get`, `list`, `watch`
+|`forklift.konveyer.io`
+|`forkliftcontrollers`
+|===
+
+.Procedure
+
+* To grant cluster-wide permissions, use `ClusterRoleBinding` to add the permissions previously described to a `ClusterRrole` and bind them to the user.
```

Comment:
the UI look for `list` permissions over namespace, so maybe we should explicitly say here that we need to grant the user list ( group is "" for namespace )

|`get`, `list`, `watch`
|
|`namespaces`

Reply:
@ahadas ^^ ?

Reply:
Ref: https://github.com/kubev2v/forklift-documentation/pull/413/files#r1317347980

Reply:
the question is whether or not that's considered common knowledge in k8s/openshift - I'd prefer not to document things that are not specific to forklift/mtv

is there anything but namespaces that we need to mention? I guess users would also need permissions for virtual machines to see the imported vms, etc

@RichardHoch did you check how the permissions are documented for OpenShift Virtualization?

Reply:
looking at the documentation of [OpenShift Virtualization](https://docs.openshift.com/container-platform/4.13/virt/virt-security-policies.html), seems they document only the cluster roles and with predefined roles - we have predefined roles in forklift as well but I think they were missing some things...

Reply:
https://docs.openshift.com/dedicated/osd_cluster_admin/osd-admin-roles.html - hmm, they talk only about dedicated `cluster-admin` role, i guess it has all the permissions needed (including the list namespace thing)

I'm ok with not documenting it specific in our docs, it's an openshift issue, if you have admin permissions in openshift ( e.g. `cluster-admin` ) the UI will let you see admin things.

Reply:
It looks like the test for listing namespace is a technical detail for how to test you have `cluster-admin`, it makes sense not to document technical details.

Reply:
@ahadas I simply made a table based on Qin's comments in the Jira. Please advise on any changes, including information, that need to be made.